packages/material-ui/src/test-utils/describeConformance.js

```diff
@@ -109,9 +109,16 @@ function describeRef(element, getOptions) {
   describe('ref', () => {
     it(`attaches the ref`, () => {
       // type def in ConformanceOptions
-      const { mount, refInstanceof } = getOptions();
+      const { inheritComponent, mount, refInstanceof } = getOptions();
 
-      testRef(element, mount, current => assert.instanceOf(current, refInstanceof));
+      testRef(element, mount, (instance, wrapper) => {
+        assert.instanceOf(instance, refInstanceof);
+
+        if (inheritComponent && instance instanceof window.Element) {
+          const rootHost = findOutermostIntrinsic(wrapper);
+          assert.strictEqual(instance, rootHost.instance());
```

Comment:
I'm not sure to fully understand the implication. Does it prevent the `Switch.js` pattern to pass the ref test? (that has a wrapping element)

Reply:
Switch is not a conform component (according to the definition you preferred) as far as I know.

Reply:
Sure, but what are we going to do with it? I think that we should break it for the InputBase component. It's probably more intuitive to farward the props to the input element. 80% of the time, users can ignore the wrapping div 🤔.

Reply:
> Sure, but what are we going to do with the Switch component?  

Why is this question important here?

Reply:
I was wondering out of my lack of knowledge of the `describeConformance.js` logic.
We move from a test that only checks that the ref is forwarded *somewhere* to a test that checks where the ref is forwarded. Now, I realize we do the same with the other tests, so 👍.

Reply:
Exactly. Should've compiled the relevant conversations into a proper PR description.